packages/ag-charts-community/src/chart/series/series.ts

```diff
@@ -420,9 +420,9 @@ export abstract class Series<
     // The union of the series domain ('community') and series-option domains ('enterprise').
     getDomain(direction: ChartAxisDirection): any[] {
         const seriesDomain: any[] = this.getSeriesDomain(direction);
-        const moduleDomains: any[][] = this.moduleMap.mapModules((module) => module.getDomain(direction));
+        const moduleDomains: any[][] = this.moduleMap.mapModules((module) => module.getDomain(direction)).flat();
         // Flatten the 2D moduleDomains into a 1D array and concatenate it with seriesDomain
-        return seriesDomain.concat(moduleDomains.flat());
+        return moduleDomains.length !== 0 ? seriesDomain.concat(moduleDomains.flat()) : seriesDomain;
```

Comment:
```suggestion
        return moduleDomains.length !== 0 ? seriesDomain.concat(moduleDomains) : seriesDomain;
```
nit: If you already flattened it you don't need to do it again.

Reply:
Thanks!